File: CableManagementStudio/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CableManagementStudio.Models
{
    public class Payment
    {
        [Key]
        public int PaymentId { get; set; }

        [Required]
        public int CustomerId { get; set; }

        public Customer? Customer { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        public DateTime PaymentDate { get; set; } = DateTime.Now;

        [Required]
        [MaxLength(50)]
        public string PaymentMode { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Paid";

        [MaxLength(255)]
        public string? Remarks { get; set; }
    }
}

File: CableManagementStudio.UI/src/app/core/models/payment.ts
export interface Payment {
  paymentId: number;

  customerId: number;

  customer?: {
    customerId: number;
    name: string;
    connectionNumber: string;
  };

  amount: number;

  paymentDate: string;

  paymentMode: string;

  status: string;

  remarks?: string;
}
